sched/clock/clock_initialize.c

```diff
@@ -155,14 +155,22 @@ int clock_basetime(FAR struct timespec *tp)
  ****************************************************************************/
 
 #ifdef CONFIG_RTC
-static void clock_inittime(void)
+static void clock_inittime(FAR const struct timespec *tp)
 {
   /* (Re-)initialize the time value to match the RTC */
 
 #ifndef CONFIG_CLOCK_TIMEKEEPING
   struct timespec ts;
 
-  clock_basetime(&g_basetime);
+  if (tp)
+    {
+      memcpy(&g_basetime, tp, sizeof(struct timespec));
```

Comment:
Maybe can go with the shorter `g_clock_wall_time = *tp;` as in `clock_timekeeping_set_wall_time` or switch to `memcpy` there. Just for style consistency.